src/pdaaal/engine/PAutomaton.cpp

```diff
@@ -151,19 +153,39 @@ namespace pdaaal {
         // This is an implementation of Algorithm 2 (figure 3.4) in:
         // Schwoon, Stefan. Model-checking pushdown systems. 2002. PhD Thesis. Technische Universität München.
         // http://www.lsv.fr/Publis/PAPERS/PDF/schwoon-phd02.pdf (page 48)
+        auto & pda_states = pda().states();
+        auto n_pda_states = pda_states.size();
 
         std::unordered_set<temp_edge_t, temp_edge_hasher> edges;
         std::stack<temp_edge_t> trans;
-        std::vector<temp_edge_t> rel;
-        auto & pda_states = pda().states();
-        auto n_pda_states = pda_states.size();
-        auto insert_edge = [&edges, &trans, &rel, this](size_t from, uint32_t label, size_t to,
+
+        // for <p, y> -> <p', y1 y2> do  (line 3)
+        //   Q' U= {q_p'y1}              (line 4)
+        std::unordered_map<std::pair<size_t, uint32_t>, size_t, boost::hash<std::pair<size_t, uint32_t>>> q_prime{};
```

Comment:
This could be replaced by 
`std::vector<std::vector<std::pair<uint32_t,size_t>>> q_prime(n_pda_states)`
for direct/faster access instead of using hashing.

Do this in "next version" rather than this pr.